File: heap/pq.java
import java.util.PriorityQueue;

public class PriorityQueueExample {
    public static void main(String[] args) {
        // Priority Queue using Java's in-built PriorityQueue class
        PriorityQueue<Integer> pq = new PriorityQueue<>();

        // Adding elements
        pq.add(10);
        pq.add(1);
        pq.add(30);

        // Peek the top element (min element in this case)
        System.out.println("Top Element: " + pq.peek());

        // Remove the top element
        System.out.println("Popped Element: " + pq.poll());

        // Peek again
        System.out.println("Top Element after pop: " + pq.peek());
    }
}

File: heap/pq.py
import heapq

# Priority Queue using heapq (Min Heap)
class PriorityQueue:
    def __init__(self):
        self._queue = []

    def push(self, item):
        heapq.heappush(self._queue, item)

    def pop(self):
        return heapq.heappop(self._queue)

    def peek(self):
        return self._queue[0] if self._queue else None

# Test Priority Queue
pq = PriorityQueue()
pq.push(10)
pq.push(1)
pq.push(30)
print("Top Element:", pq.peek())
print("Popped Element:", pq.pop())
print("Top Element after pop:", pq.peek())
